manubot/cite/citations.py

```diff
@@ -177,17 +178,36 @@ def csl_json(self) -> str:
         json_str += "\n"
         return json_str
 
-    def write_csl_json(self, path):
-        """
-        Write CSL Items to a JSON file at `path`.
+    @property
+    def csl_yaml(self) -> str:
+        from manubot.util import get_configured_yaml
+
+        yaml = get_configured_yaml()
+        assert hasattr(self, "csl_items")
+        # dump rather than safe_dump is required for
+        # pyyaml to use custom representers.
+        return yaml.dump(
+            data=self.csl_items,
+            default_flow_style=False,
+            width=float("inf"),
+            allow_unicode=True,
+            sort_keys=False,
+        )
+
+    def write_csl_items(self, path: tp.Union[os.PathLike, str, None]) -> None:
+        """
+        Write CSL Items to a JSON or YAML file at `path`.
+        If path ends with a .yml or .yaml extension, write as CSL YAML.
```

Comment:
I like this strategy for YAML support.  Continue using JSON by default and support YAML optionally based on extension.